src/utils.js

```diff
@@ -15,3 +20,68 @@ module.exports = function goToPath(path, obj) {
   }
   return current;
 };
+
+
+/**
+ * N.B. baseGetTag(), isObject() & isFunction() are lifted straight out of Lodash,
+ * with a little tweaking to conform to the coding standards in this project
+ *
+ * Lodash is awesome but I did not want add an extra dependency to this project!
+ */
+
+/**
+ * The base implementation of `getTag` without fallbacks for buggy environments.
+ *
+ * @private
+ * @param {*} value The value to query.
+ * @returns {string} Returns the `toStringTag`.
+ */
+function baseGetTag(value) {
+  if (value == null) {
```

Comment:
this is lifted straight from lodash, the `==` check makes sense in this context.